frontend/src/components/smart/CharacterDisplay.vue

```diff
@@ -4,41 +4,69 @@
           <div class="btn-trigger" @click="hideSideBar(!toggled)">
             <img :class="!toggled ? 'rotateLeft' : 'rotateRight'" src="../../assets/left-arrow.png" alt="">
           </div>
-          <b-col class="character-list"
+          <b-col class="character-list" v-if="currentPath != '/blacksmith'"
               v-bind:class="[getIsInCombat ? 'disabled-li' : '', getIsCharacterViewExpanded ? '' : 'centered-list']">
                 <div
                   :class="`${setListClassForSelChar(c.id, currentCharacterId)}`"
                   :style="`--staminaReady: ${(getCharacterStamina(c.id)/maxStamina)*100}%;`"
                   v-for="c in filteredCharactersForList"
                   :key="c.id"
                   @click="!getIsInCombat && setCurrentCharacter(c.id) && alert(c.id)"
+                >
+                  <div class="character-element">
+                    <div class="element-frame">
+                        <div>
+                          <span :id="`${setIdForElement(c.trait, c.isSelected)}`"/>
+                        </div>
+                    </div>
+                    <div class="element-frame-active">
+                      <div>
+                          <span
+                            :id="`${setIdForElement(c.trait, c.isSelected)}`"
+                          />
+                        </div>
+                    </div>
+                  </div>
+                  <div class="character-details">
+                    <div class="name-list">
+                      {{ getCleanCharacterName(c.id) }}
+                    </div>
+                    <div class="small-stamina-char"
+                      :style="`--staminaReady: ${(getCharacterStamina(c.id)/maxStamina)*100}%;`"
+                      v-tooltip.bottom="toolTipHtml(timeUntilCharacterHasMaxStamina(c.id))">
+                    </div>
+                    <div class="char-level">
+                        {{$t('PlayToEarn.level')}} {{ c.level + 1}} <span> (STA {{ getCharacterStamina(c.id) }} / 200)</span>
+                    </div>
+                  </div>
+                </div>
+          </b-col>
+          <b-col class="character-list centered-list" v-else>
+                <div v-for="s in sideBarBlacksmith"
```

Comment:
s is too verbose can you make it clear what you are looping on at a glance it is not clear what it is.